cli/pkg/local/app.go

```diff
@@ -182,9 +184,12 @@ func NewApp(ctx context.Context, opts *AppOptions) (*App, error) {
 		"rill.download_row_limit": "0", // 0 means unlimited
 		"rill.stage_changes":      "false",
 	}
-	for k, v := range opts.Variables {
-		vars[k] = v
+	maps.Copy(vars, opts.Variables)
+	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
```

Comment:
**Issue**: Directly reading the .env file within the "NewApp" function increases coupling and reduces testability. It also handles the file path construction inline, which can lead to errors if not handled carefully. <br> **Fix**: Refactor the environment loading logic into a separate function that can be easily tested and reused. Also, ensure that the file path handling is robust against common errors. <br> **Code Suggestion**: 
 ```
 - envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
 + envMap, err := loadEnvFile(projectPath)

 // Somewhere in the same file
 + func loadEnvFile(projectPath string) (map[string]string, error) {
 +   path := filepath.Join(projectPath, ".env")
 +   return godotenv.Read(path)
 + }
 ```